tasks/README.md

```diff
@@ -80,7 +79,11 @@ The preferred style of the keys should be `snake_case`.
       "description": "Description of optional param3",
       "type": "Optional[Integer]"
     }
-  }
+  },
+  "implementations" : [
+    {"name": "foo.sh", "requirements": ["shell"]},
+    {"name": "foo.ps1", "requirements": ["powershell"]}
```

Comment:
is this a case of first to match for a machine will be used? I imagine there could be overlap in requirements.  Do you have some kind of most suitable situation or just first to match?

Does `puppet-agent` imply that you might require its ruby? should we be concerned with versions?

In MCollective we found it necessary to set requirements with versions because in the long run even the basic thing like `puppet-agent` might change, say a task requires ruby features only available in the ruby provided by agent 6 and not 5 that seems like a real situation we might hit.

Is it really a case of tasks being able to just specify any old stuff here and a runner could decide to make sense of it or not? That seems prone to simple error - I accidentally set `puppet-agetn` as a requirement and it wont complain it will just silently not do what I intend exactly? This free form stuff makes me worry about early user feedback on error and discoverability of errors

Reply:
@adreyer thanks, any comment about the free form nature of these?

Reply:
It seems like that would be quickly apparent. We can probably make it more so by including what features were present when no suitable implementation is found. Generating a warning in pdk seems feasible too